src/components/GameItem.js

```diff
@@ -4,26 +4,64 @@ import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
 class GameItem extends Component {
-  propTypes = {
-    height: PropTypes.number.isRequired,
-    layer: PropTypes.number.isRequired,
+  constructor(props) {
+    super(props);
+    this.state = {
+      checked: false,
+      divClass: 'game-item'
+    };
   }
+  onItemClicked = () => {
+    console.log(this);
+    if (this.state.clicked) {
+      this.setState({ clicked: false });
+    }
+  };
+
+  gotIt = () => {
+    if (!this.state.checked) {
+      this.setState({
+        checked: true
+      });
+
+      if (this.props.type === 'litter') {
+        this.props.clickCallback(this.props.type);
+        this.setState({
+          divClass: 'game-item spotted-litter'
+        });
+      } else {
+        this.props.clickCallback(this.props.type);
+        this.setState({
+          divClass: 'game-item spotted-nature'
+        });
+      }
+    }
+  };
 
   render() {
     const itemStyle = {
       bottom: `${this.props.height}px`, // use props.height to offset from the bottom of screen
-      zIndex: this.props.layer, // use props.layer to set z-index, so we display ontop of background
+      zIndex: this.props.layer // use props.layer to set z-index, so we display ontop of background
     };
 
-    // Update this to select the correct icon for each item
-    const icon = ItemIcons.rock;
+    const icon = ItemIcons[`${this.props.type}`];
```

Comment:
If you're interested in alternate syntax, you can actually just do `const icon = ItemIcons[this.props.type];` here!

Reply:
Ooooh. Nice :) Thanks, Dee!